src/games/gameitem.py

```diff
@@ -169,26 +169,32 @@ def update(self, message, old_client=None):
             logger.error('gamesitem.update called with 3 args')
             logger.error(traceback.format_stack())
 
-        if self.title is None:  # you can't change this...
+        if self.title is None:  # new game hosted
             self.title = message['title']
-            self.host = message['host']
             self.password_protected = message.get('password_protected', False)
             self.mod = message['featured_mod']
 
-            if 'host_id' in message:
-                self.hostid = message['host_id']
-            else:
-                self.hostid = client.instance.players.getID(self.host)
+        if self.host is None:  # new game hosted
+            self.host = message['host']
+        elif self.host != message['host']:  # somethings funny (offline)
+            self.host = message['host']
 
-        # Maps integral team numbers (from 2, with 1 "none") to lists of names.
-        teams_map = dict.copy(message['teams'])
-        self.modVersion = message.get('featured_mod_versions', [])
-        self.mods = message.get('sim_mods', {})
-        self.options = message.get('options', [])
-        num_players = message.get('num_players', 0)
-        self.slots = message.get('max_players', 12)
-        
-        oldstate = self.state
+        if 'host_id' in message:
+            self.hostid = message['host_id']
+        elif self.hostid == -1:  # fresh game
+            self.hostid = client.instance.players.getID(self.host)
+        elif self.hostid != client.instance.players.getID(self.host):  # somethings funny (offline)
+            self.hostid = client.instance.players.getID(self.host)
```

Comment:
I really don't like this defensive programming, what's the purpose of doing exactly the same thing in the 'else' branch?